src/debugger/launch.ts

```diff
@@ -113,14 +114,15 @@ function createExecutableConfigurations(ctx: FolderContext): vscode.DebugConfigu
     const executableProducts = ctx.swiftPackage.executableProducts;
     const { folder, nameSuffix } = getFolderAndNameSuffix(ctx);
     const buildDirectory = BuildFlags.buildDirectoryFromWorkspacePath(folder, true);
+    const binaryExtension = process.platform === "win32" ? ".exe" : "";
     return executableProducts.flatMap(product => {
         return [
             {
                 type: DebugAdapter.adapterName,
                 request: "launch",
                 sourceLanguages: ["swift"],
                 name: `Debug ${product.name}${nameSuffix}`,
-                program: `${buildDirectory}/debug/` + product.name,
+                program: path.join(buildDirectory, "debug", product.name + binaryExtension),
```

Comment:
```suggestion
                program: path.join(buildDirectory, "debug", `${product.name}${binaryExtension}`),
```

Reply:
Extracting the construction seems better:

```javascript
const Configuration = Object.freeze({
  Debug: "debug",
  Release: "release",
})

function product(configuration, name) {
  const extension = process.platform === "win32" ? ".exe" : ""
  return path.join(buildDirectory, Configuration[configuration], `${name}${extension}`)
}
```

This would give us something like:
```javascript
product(Configuration.Release, "test")
```